File: frontend/src/lib/config/site.ts
import { writable, get } from 'svelte/store';
import type { PublicSiteConfig } from '$lib/types/siteConfig';

export type SiteConfig = PublicSiteConfig & {
	logo_dark_path: string;
	logo_light_path: string;
};

const DEFAULTS: SiteConfig = {
	site_name: 'Afterglow',
	site_description: 'OpenStack VM + OverlayFS 배포 플랫폼',
	logo_path: '/logo.png',
	logo_dark_path: '/logo-dark.png',
	logo_light_path: '/logo-white.png',
	favicon_path: '/favicon.ico',
	refresh_interval_ms: 5000,
	services: { magnum: false, manila: false, zun: false, k3s: false, trove: false, swift: false, barbican: false },
	runtime: {
		api_base: '',
		s3_base: '',
		grafana_base: '',
	},
};

export const siteConfig = writable<SiteConfig>({ ...DEFAULTS });

export function initSiteConfig(config: PublicSiteConfig): void {
	siteConfig.set({
		...DEFAULTS,
		...config,
		services: { ...DEFAULTS.services, ...config.services },
		runtime: { ...DEFAULTS.runtime, ...config.runtime },
	});
}

export function getSiteName(): string {
	return get(siteConfig).site_name;
}

export function getSiteDescription(): string {
	return get(siteConfig).site_description;
}

File: backend/app/api/common/site.py
from fastapi import APIRouter

from app.config import get_settings

router = APIRouter()


@router.get("")
def get_site_config():
    """사이트 표시 이름 및 설명 반환 (인증 불필요)."""
    s = get_settings()
    return {
        "site_name": s.site_name,
        "site_description": s.site_description,
        "logo_path": s.logo_path,
        "favicon_path": s.favicon_path,
        "services": {
            "magnum": s.service_magnum_enabled,
            "manila": s.service_manila_enabled,
            "zun": s.service_zun_enabled,
            "k3s": s.service_k3s_enabled,
        },
    }
